pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -172,13 +170,23 @@ func (r *reconciler) reportEvent(project *gardencorev1beta1.Shoot, eventType str
 }
 
 // determineSeed returns an appropriate Seed cluster (or nil).
-func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.SeedLister, shootLister gardencorelisters.ShootLister, cloudProfileLister gardencorelisters.CloudProfileLister, strategy config.CandidateDeterminationStrategy) (*gardencorev1beta1.Seed, error) {
+func determineSeed(
+	ctx context.Context,
+	gardenClient client.Client,
+	shoot *gardencorev1beta1.Shoot,
+	seedLister gardencorelisters.SeedLister,
+	cloudProfileLister gardencorelisters.CloudProfileLister,
```

Comment:
Can we also switch to a c-r clients here?
It seems like this should be easily doable.
Thereby we could avoid confusion and get rid of the remaining `Lister`/`SharedInformerFactory` usage.

Reply:
Yeah, I didn't do this because the `CachedRuntimeClient` feature gate is alpha and disabled by default, i.e., we would never have watches and do direct calls here.
OTOH, we could either

1. remove the listers and otherwise do nothing for now and live with the direct calls (the GSCHED usually runs next to GAPI, hence, the calls should at least not contribute to network I/O, and also GSCHED is not too busy because usually not a lot of a shoots get created), or
1. explicitly start the C-R informers to have WATCHes for `CloudProfile`s and `Seed`s.

What do you prefer?

Reply:
> Yeah, I didn't do this because the `CachedRuntimeClient` feature gate is alpha and disabled by default, i.e., we would never have watches and do direct calls here.

True of course. We could still explicitly get/list from the cache which is started in both cases.

It will be more tricky for gcm and gardenlet as we will then cause duplicated watches, but for the scheduler (where its an isolated controller) we should be able to easily tackle the problem this way.

I'm basically fine with either way, but slightly prefer explicitly listing from the cache.

Reply:
OK, I've pushed 4480443 now, I hope this fits to your expectations :)